services/watson-extension/src/watson_extension/clients/platform/integrations.py

```diff
@@ -0,0 +1,203 @@
+from typing import Optional, Tuple, List, Dict
+import abc
+import injector
+import logging
+import enum
+from urllib.parse import urlparse
+from aiohttp import ClientResponse
+from watson_extension.clients import PlatformNotificationsURL
+from watson_extension.clients.identity import AbstractUserIdentityProvider
+from watson_extension.clients.platform_request import AbstractPlatformRequest
+from watson_extension.clients.platform import IntegrationInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+reporting_types = ["ansible"]
+communications_camel_subtypes = ["google_chat", "teams", "slack"]
+reporting_camel_subtypes = ["servicenow", "splunk"]
+MAX_NUMBER_OF_INTEGRATIONS = 5
+
+
+class CreateEndpointResponseType(enum.Enum):
+    OK = "ok"
+    NOT_HTTPS = "not_https"
+    ATTRIBUTION_ERROR = "attribution_error"
+
+
+async def validate_integration_url(
+    integrations_setup_url: str,
+) -> CreateEndpointResponseType:
+    try:
+        result = urlparse(integrations_setup_url)
+        if result.scheme != "https":
+            return CreateEndpointResponseType.NOT_HTTPS
+    except AttributeError:
+        return CreateEndpointResponseType.ATTRIBUTION_ERROR
+
+    return CreateEndpointResponseType.OK
+
+
+def notifications_type_to_group(integration: Dict) -> str:
+    if integration["type"] == "webhook":
+        return "webhook"
+    if integration["type"] in reporting_types:
+        return "reporting"
+
+    if integration["type"] == "camel":
+        if integration["sub_type"] in communications_camel_subtypes:
+            return "communications"
+        if integration["sub_type"] in reporting_camel_subtypes:
+            return "reporting"
+
+
+class IntegrationsClient(abc.ABC):
+    @abc.abstractmethod
+    async def create_endpoint(
+        self,
+        setup_name: str,
+        setup_url: str,
+        secret_token: str,
+        setup_type: str,
+        setup_sub_type: Optional[str] = None,
+    ) -> Tuple[bool, CreateEndpointResponseType]: ...
+
+    @abc.abstractmethod
+    async def fetch_integrations(
+        self, search: Optional[str] = None, enabled: Optional[bool] = None
+    ) -> Tuple[bool, Optional[List[IntegrationInfo]]]: ...
+
+    @abc.abstractmethod
+    async def integration_unpause(self, integration_id: str) -> ClientResponse: ...
```

Comment:
Nitpicky, could these be pause_integration and resume_integration (or unpause is fine haha)

Reply:
I ended up using `integration_resume`, we have a similar method on the Sources client. They will be used together in the Core/Rotues.